OLA RIDE ANALYTICS PROJECT

In [1]:
## 1.Import Required Libraries
import pandas as pd
import numpy as np

In [2]:
# Display all columns
pd.set_option("display.max_columns", None)

In [3]:
# Display wider tables
pd.set_option("display.width", 1000)

In [9]:
## 2. Load Dataset
df = pd.read_csv(r"C:\Users\Vishal Dokhe\Downloads\Bengaluru Ola.csv")

In [11]:
# Basic Information
print("Dataset Loaded Successfully!\n")
print(f"Rows    : {df.shape[0]}")
print(f"Columns : {df.shape[1]}")

Dataset Loaded Successfully!

Rows    : 49999
Columns : 21


In [12]:
df.head()

,Date,Time,Booking ID,Booking Status,Customer ID,Vehicle Type,Pickup Location,Drop Location,Avg VTAT,Avg CTAT,Cancelled by Customer,Reason for Cancelling by Customer,Cancelled Rides by Driver,Reason for Cancelling by Driver,Incomplete Rides,Incomplete Rides Reason,Booking Value,Payment Method,Ride Distance,Driver Ratings,Customer Rating
0,28/01/2024,6:00:00,CNR1721175,Success,329258,Auto,Area-3,Area-2,5.42,18.46,0,NaN,0,NaN,0,NaN,868.06,Wallet,28.5,4.4,4.4
1,26/01/2024,3:00:00,CNR2871422,Cancelled by Driver,201414,Mini,Area-7,Area-6,NaN,NaN,0,NaN,1,The customer was coughing/sick,0,NaN,NaN,NaN,NaN,NaN,NaN
2,15/01/2024,16:00:00,CNR6875935,Cancelled by Driver,301629,Bike,Area-40,Area-24,NaN,NaN,0,NaN,1,Personal & Car related issues,0,NaN,NaN,NaN,NaN,NaN,NaN
3,02/01/2024,22:00:00,CNR6798834,Cancelled by Driver,319684,Prime Sedan,Area-11,Area-24,NaN,NaN,0,NaN,1,The customer was coughing/sick,0,NaN,NaN,NaN,NaN,NaN,NaN
4,30/01/2024,22:00:00,CNR9661713,Incomplete,330283,Bike,Area-41,Area-45,NaN,NaN,0,NaN,0,NaN,1,Other Issue,NaN,NaN,NaN,NaN,NaN


In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49999 entries, 0 to 49998
Data columns (total 21 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   Date                               49999 non-null  object 
 1   Time                               49999 non-null  object 
 2   Booking ID                         49999 non-null  object 
 3   Booking Status                     49999 non-null  object 
 4   Customer ID                        49999 non-null  int64  
 5   Vehicle Type                       49999 non-null  object 
 6   Pickup Location                    49999 non-null  object 
 7   Drop Location                      49999 non-null  object 
 8   Avg VTAT                           33484 non-null  float64
 9   Avg CTAT                           33484 non-null  float64
 10  Cancelled  by Customer             49999 non-null  int64  
 11  Reason for Cancelling by Customer  3799 non-null   obj

In [14]:
df.describe(include="all")

,Date,Time,Booking ID,Booking Status,Customer ID,Vehicle Type,Pickup Location,Drop Location,Avg VTAT,Avg CTAT,Cancelled by Customer,Reason for Cancelling by Customer,Cancelled Rides by Driver,Reason for Cancelling by Driver,Incomplete Rides,Incomplete Rides Reason,Booking Value,Payment Method,Ride Distance,Driver Ratings,Customer Rating
count,49999,49999,49999,49999,49999.000000,49999,49999,49999,33484.000000,33484.000000,49999.000000,3799,49999.000000,9610,49999.000000,3106,33484.000000,33484,33484.000000,33484.00000,33484.000000
unique,31,24,49866,4,NaN,7,50,50,NaN,NaN,NaN,5,NaN,4,NaN,3,NaN,4,NaN,NaN,NaN
top,12/01/2024,0:00:00,CNR1569898,Success,NaN,Prime Plus,Area-39,Area-39,NaN,NaN,NaN,Driver is not moving towards pickup location,NaN,More than permitted people in there,NaN,Customer Demand,NaN,Cash,NaN,NaN,NaN
freq,1766,2171,2,33484,NaN,7252,1100,1058,NaN,NaN,NaN,793,NaN,2512,NaN,1091,NaN,8552,NaN,NaN,NaN
mean,NaN,NaN,NaN,NaN,550979.937539,NaN,NaN,NaN,10.481654,15.564320,0.075982,NaN,0.192204,NaN,0.062121,NaN,1023.374286,NaN,25.447143,4.00304,4.001726
std,NaN,NaN,NaN,NaN,258855.930945,NaN,NaN,NaN,5.490853,8.365313,0.264971,NaN,0.394036,NaN,0.241378,NaN,563.595072,NaN,14.175953,0.57968,0.579177
min,NaN,NaN,NaN,NaN,100001.000000,NaN,NaN,NaN,1.000000,1.000000,0.000000,NaN,0.000000,NaN,0.000000,NaN,50.100000,NaN,1.000000,3.00000,3.000000
25%,NaN,NaN,NaN,NaN,328740.500000,NaN,NaN,NaN,5.730000,8.310000,0.000000,NaN,0.000000,NaN,0.000000,NaN,532.942500,NaN,13.100000,3.50000,3.500000
50%,NaN,NaN,NaN,NaN,552598.000000,NaN,NaN,NaN,10.460000,15.610000,0.000000,NaN,0.000000,NaN,0.000000,NaN,1023.350000,NaN,25.460000,4.00000,4.000000
75%,NaN,NaN,NaN,NaN,773970.000000,NaN,NaN,NaN,15.240000,22.790000,0.000000,NaN,0.000000,NaN,0.000000,NaN,1514.677500,NaN,37.670000,4.50000,4.500000


In [15]:
df.isnull().sum()

Date                                     0
Time                                     0
Booking ID                               0
Booking Status                           0
Customer ID                              0
Vehicle Type                             0
Pickup Location                          0
Drop Location                            0
Avg VTAT                             16515
Avg CTAT                             16515
Cancelled  by Customer                   0
Reason for Cancelling by Customer    46200
Cancelled Rides by Driver                0
Reason for Cancelling by Driver      40389
Incomplete Rides                         0
Incomplete Rides Reason              46893
Booking Value                        16515
Payment Method                       16515
Ride Distance                        16515
Driver Ratings                       16515
Customer Rating                      16515
dtype: int64

In [16]:
df.duplicated().sum()

np.int64(0)

In [17]:
df["Booking Status"].value_counts()

Booking Status
Success                  33484
Cancelled by Driver       9610
Cancelled by Customer     3799
Incomplete                3106
Name: count, dtype: int64

In [19]:
df["Vehicle Type"].value_counts()

Vehicle Type
Prime Plus     7252
Bike           7223
Prime Sedan    7179
Prime SUV      7140
Auto           7098
eBike          7097
Mini           7010
Name: count, dtype: int64

In [20]:
df["Payment Method"].value_counts(dropna=False)

Payment Method
NaN       16515
Cash       8552
UPI        8428
Card       8280
Wallet     8224
Name: count, dtype: int64

In [21]:
df["Reason for Cancelling by Customer"].value_counts(dropna=False)

Reason for Cancelling by Customer
NaN                                             46200
Driver is not moving towards pickup location      793
AC is not working                                 792
Change of plans                                   756
Driver asked to cancel                            738
Wrong Address                                     720
Name: count, dtype: int64

In [22]:
df["Reason for Cancelling by Driver"].value_counts(dropna=False)

Reason for Cancelling by Driver
NaN                                    40389
More than permitted people in there     2512
Personal & Car related issues           2413
Customer related issue                  2402
The customer was coughing/sick          2283
Name: count, dtype: int64

In [23]:
df["Incomplete Rides Reason"].value_counts(dropna=False)

Incomplete Rides Reason
NaN                  46893
Customer Demand       1091
Other Issue           1029
Vehicle Breakdown      986
Name: count, dtype: int64

In [24]:
df.dtypes

Date                                  object
Time                                  object
Booking ID                            object
Booking Status                        object
Customer ID                            int64
Vehicle Type                          object
Pickup Location                       object
Drop Location                         object
Avg VTAT                             float64
Avg CTAT                             float64
Cancelled  by Customer                 int64
Reason for Cancelling by Customer     object
Cancelled Rides by Driver              int64
Reason for Cancelling by Driver       object
Incomplete Rides                       int64
Incomplete Rides Reason               object
Booking Value                        float64
Payment Method                        object
Ride Distance                        float64
Driver Ratings                       float64
Customer Rating                      float64
dtype: object

# Observations

- The dataset contains 49,999 ride records and 21 columns.
- No duplicate records were found.
- The dataset includes four booking statuses: Success, Cancelled by Driver, Cancelled by Customer, and Incomplete.
- Missing values are expected because unsuccessful rides do not contain completed ride information such as booking value, payment method, ride distance, and ratings.
- Vehicle types are well distributed across the dataset.
- Cancellation reasons are clean and standardized.
- The dataset is in good condition and requires only minimal preprocessing before analysis.